examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldServerS2A.java

```diff
@@ -0,0 +1,141 @@
+        this.privateKeyFilePath = privateKeyFilePath;
+        this.trustCertCollectionFilePath = trustCertCollectionFilePath;
+    }
+
+    private SslContextBuilder getSslContextBuilder() {
+      SslContextBuilder sslClientContextBuilder = SslContextBuilder.forServer(new File(certChainFilePath), new File(privateKeyFilePath));
+        sslClientContextBuilder.trustManager(new File(trustCertCollectionFilePath));
+        sslClientContextBuilder.clientAuth(ClientAuth.REQUIRE);
+        sslClientContextBuilder.protocols("TLSv1.3");
+        return GrpcSslContexts.configure(sslClientContextBuilder);
+    }
+
+    private void start() throws IOException {
+        server = NettyServerBuilder.forPort(port)
+                .sslContext(getSslContextBuilder().build())
+                .addService(new GreeterImpl())
+                .build()
+                .start();
+
+        logger.info("Server started, listening at address " + port);
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                // Use stderr here since the logger may have been reset by its JVM shutdown hook.
+                System.err.println("shutting down gRPC server since JVM is shutting down");
+                HelloWorldServerS2A.this.stop();
+                System.err.println("server shut down");
+            }
+        });
+    }
+
+    private void stop() {
+        if (server != null) {
+            server.shutdown();
+        }
+    }
+
+    /**
+     * Awaits termination on the main thread since the grpc library uses daemon threads.
+     */
+    private void blockUntilShutdown() throws InterruptedException {
+        if (server != null) {
+            server.awaitTermination();
+        }
+    }
+
+    private static HelloWorldServerS2A create(int port,
+                             String certChainFilePath,
+                             String privateKeyFilePath,
+                             String trustCertCollectionFilePath) {
+        checkNotNull(port, "port should not be Null");
+        checkNotNull(certChainFilePath, "certChainFilePath should not be Null");
+        checkNotNull(privateKeyFilePath, "privateKeyFilePath should not be Null");
+        checkNotNull(trustCertCollectionFilePath, "trustCertCollectionFilePath should not be Null");
+        return new HelloWorldServerS2A(port,
+            certChainFilePath,
+            privateKeyFilePath,
+            trustCertCollectionFilePath);
+    }
+
+    /**
+     * Main launches the server from the command line.
+     *
+     * args:
+     * [port]
+     * [certChainFilePath] File Path to Server Certificate
+     * [privateKeyFilePath] File Path to Server Private Key
+     * [trustCertCollectionFilePath] File Path to CA Certificate
+     */
+    public static void main(String[] args) throws IOException, InterruptedException {
+        HelloWorldServerS2A server = HelloWorldServerS2A.create(
```

Comment:
As in the client file, please have default values that are overridden by the user's flags.